File: backend/src/main/java/com/github/ryand6/sudokuweb/enums/PreferenceDirection.java
package com.github.ryand6.sudokuweb.enums;

import lombok.Getter;

@Getter
public enum PreferenceDirection {

    LESS("Less", "Shorter", "Easier"),
    EQUAL("Same", "Same", "Same"),
    MORE("More", "Longer", "Harder");

    private final String genericLabel;
    private final String durationLabel;
    private final String difficultyLabel;

    PreferenceDirection(String genericLabel, String durationLabel, String difficultyLabel) {
        this.genericLabel = genericLabel;
        this.durationLabel = durationLabel;
        this.difficultyLabel = difficultyLabel;
    }

}

File: frontend/src/types/enum/PreferenceDirection.ts
export type PreferenceDirection = "LESS" | "EQUAL" | "MORE";

export const PreferenceDirectionLabels: Record<PreferenceDirection, {
    generic: string,
    duration: string,
    difficulty: string
}> = {
    LESS : {generic: "Less", duration: "Shorter", difficulty: "Easier"},
    EQUAL: {generic: "Same", duration: "Same", difficulty: "Same"},
    MORE: {generic: "More", duration: "Longer", difficulty: "Harder"}
}